src/main/java/com/github/manolo8/darkbot/gui/drawables/DynamicEntitiesDrawer.java

```diff
@@ -131,23 +146,61 @@ private void drawNpcs(MapGraphics mg) {
     }
 
     private void drawPlayers(MapGraphics mg) {
+        Set<Integer> oosMembers = group.getMembers().stream()
+                .map(PartialGroupMember::getId)
+                .collect(Collectors.toSet());
+
+        Color ally = mg.getColor("allies");
+        Color enemy = mg.getColor("enemies");
+        Color groupMember = mg.getColor("group_member");
+
         for (Player player : players) {
+            oosMembers.remove(player.getId());
             boolean isEnemy = player.getEntityInfo().isEnemy();
-            mg.setColor(isEnemy ? "enemies" : "allies");
-            drawEntity(mg, player, 4, false);
 
+            int level = -1;
+            Color color = isEnemy ? enemy : ally;
+            if (group.hasGroup()) {
+                GroupMember member = group.getMember(player);
+                if (member != null) {
+                    color = groupMember;
+                    level = member.getLevel();
+                }
+            }
+
+            mg.setColor(color);
+            drawEntity(mg, player, 4, false);
             if (mg.hasDisplayFlag(DisplayFlag.USERNAMES))
-                mg.drawString(player, player.getEntityInfo().getUsername(), -5, MapGraphics.StringAlign.MID);
+                mg.drawString(player, player.getEntityInfo().getUsername()
+                        + (level != -1 ? " [" + level + "]" : ""), -6, MapGraphics.StringAlign.MID);
+        }
+
+        if (group.hasGroup()) {
+            mg.setColor(groupMember.darker());
+
+            Map<TextAttribute, Integer> attrs = Map.of(TextAttribute.UNDERLINE, TextAttribute.UNDERLINE_LOW_DASHED);
```

Comment:
nah, underline on the name itself ain't it, that makes them seem more important

i'd just keep the .darker on the color, and maybe make it so player entities are filled-in but these are not-filled?